templates/react/src/example/Example.tsx

```diff
@@ -0,0 +1,115 @@
+import { V3_SWAP_ROUTER_ADDRESS } from '../libs/addresses'
+import {
+  connectWallet,
+  getMainnetProvider,
+  getProvider,
+  getWalletAddress,
+  sendTransaction,
+  TransactionState,
+} from '../libs/provider'
+
+async function route(): Promise<TransactionState> {
+  const router = new AlphaRouter({ chainId: ChainId.MAINNET, provider: getMainnetProvider() })
+  const address = getWalletAddress()
+
+  if (!address) {
+    return TransactionState.Failed
+  }
+
+  const route = await router.route(
+    CurrencyAmount.fromRawAmount(
+      CurrentConfig.currencies.tokenIn,
+      ethers.utils.parseEther(CurrentConfig.currencies.amountIn.toString()).toString()
+    ),
+    CurrentConfig.currencies.tokenOut,
+    TradeType.EXACT_INPUT,
+    {
+      recipient: CurrentConfig.wallet.address,
+      slippageTolerance: new Percent(5, 100),
+      deadline: Math.floor(Date.now() / 1000 + 1800),
+      type: SwapType.SWAP_ROUTER_02,
+    }
+  )
+
+  const res = await sendTransaction({
+    data: route?.methodParameters?.calldata,
+    to: V3_SWAP_ROUTER_ADDRESS,
+    value: route?.methodParameters?.value,
+    from: address,
+    gasLimit: 300000,
+  })
+
+  return res
+}
+
+function Example() {
+  const [tokenInBalance, setTokenInBalance] = useState<string>()
+  const [tokenOutBalance, setTokenOutBalance] = useState<string>()
+  const [txState, setTxState] = useState<TransactionState>(TransactionState.New)
+  const [blockNumber, setBlockNumber] = useState<number>(0)
+
+  // Event Handlers
+
+  const onConnectWallet = async () => {
+    await connectWallet()
+    if (getWalletAddress()) {
+      refreshBalances()
+    }
+  }
+
+  const onTrade = async () => {
+    setTxState(TransactionState.Sending)
+    setTxState(await route())
+  }
+
+  // Update wallet state given a block number
+  const refreshBalances = async () => {
+    const provider = getProvider()
+    const address = getWalletAddress()
+    if (address && provider) {
+      setTokenInBalance(await getCurrencyBalance(provider, address, CurrentConfig.currencies.tokenIn))
+      setTokenOutBalance(await getCurrencyBalance(provider, address, CurrentConfig.currencies.tokenOut))
+    }
+  }
+
+  // Listen for new blocks and update the wallet
+  useEffect(() => {
+    const subscription = getProvider()?.on('block', async (blockNumber: number) => {
+      refreshBalances()
+      setBlockNumber(blockNumber)
+    })
+    return () => {
+      subscription?.removeAllListeners()
+    }
+  }, [])
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        {CurrentConfig.env === Environment.WALLET_EXTENSION && getProvider() === null && (
+          <h1 className="error">Please install a wallet to use this example configuration</h1>
+        )}
+        <h3>{`Wallet Address: ${getWalletAddress()}`}</h3>
+        {CurrentConfig.env === Environment.WALLET_EXTENSION && (
+          <button onClick={onConnectWallet}>Connect Wallet</button>
```

Comment:
We disable this if the wallet is connected.

Reply:
I left this as you may want to connect a new wallet if you've switched accounts etc - thoughts there?

Reply:
That would be awesome, but this does not seem to work the way we have it setup when I'm switching accounts.